File: 2-1_DataStructure/KSH/DS_rectangle/DS_assignment/rectangle.cpp
#include "rectangle.h"

Rectangle::Rectangle() {
	width = 0;
	height = 0;
}
Rectangle::Rectangle(int w, int h) :width(w), height(h) {
	if (width < 0) {
		width = 0;	
	}
	else {
		width = width;
	}
	if (height < 0) {
		height = 0;
	}
	else {
		height = height;
	}
}

int Rectangle::area() {
	return width * height;
}

int Rectangle::perimeter() {
	return (width + height) * 2;
}

File: 2-1_ObjectOrientedProgramming/JWC/Assignment3/Question4/Rectangle.java
public class Rectangle {
  private int x, y, width, height;

  Rectangle() {
  }

  Rectangle(int x, int y, int width, int height) {
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
  }

  public int getX() {
    return this.x;
  }

  public int getY() {
    return this.y;
  }

  public int getWidth() {
    return this.width;
  }

  public int getHeight() {
    return this.height;
  }

  public int square() {
    return width * height;
  }

  public void show() {
    System.out.println("(" + x + "," + y + ") 에서 크기가 " + width + "x" +
        height + "인 사각형");
  }

  public boolean contains(Rectangle comparedSquare) {
    if (x < comparedSquare.getX() && y < comparedSquare.getY()) {
      int endX = x + width, endY = y + height;
      int comparedSquareEndX = comparedSquare.getX() + comparedSquare.getWidth();
      int comparedSquareEndY = comparedSquare.getY() + comparedSquare.getHeight();

      if (comparedSquareEndX < endX && comparedSquareEndY < endY)
        return true;
      else
        return false;
    } else {
      return false;
    }
  }
}
